packages/sdk/typescript/human-protocol-sdk/src/staking.ts

```diff
@@ -0,0 +1,362 @@
+   * @throws {Error} - An error object if an error occurred, void otherwise
+   */
+  public async distributeRewards(escrowAddress: string): Promise<void> {
+    if (!ethers.utils.isAddress(escrowAddress)) {
+      throw ErrorInvalidEscrowAddressProvided;
+    }
+
+    try {
+      const rewardPoolContract: RewardPool = RewardPool__factory.connect(
+        await this.stakingContract.rewardPool(),
+        this.signerOrProvider
+      );
+
+      await rewardPoolContract.distributeReward(escrowAddress);
+      return;
+    } catch (e) {
+      throw ErrorStakingFailedToDistributeRewards;
+    }
+  }
+
+  /**
+   * **Returns the staking information about an staker address.*
+   *
+   * @param {string} staker - Address of the staker
+   * @returns {Promise<IStakerInfo>} - Return staking information of the specified address
+   * @throws {Error} - An error object if an error occurred, result otherwise
+   */
+  public async getStaker(staker: string): Promise<IStaker> {
+    if (!ethers.utils.isAddress(staker)) {
+      throw ErrorInvalidStakerAddressProvided;
+    }
+
+    try {
+      const result = await this.stakingContract.getStaker(staker);
+      return result;
+    } catch (e) {
+      console.error(e);
+      throw ErrorStakingGetStaker;
+    }
+  }
+
+  /**
+   * **Returns the staking information about all stakers of the protocol.*
+   *
+   * @returns {Promise<IStakerInfo>} - Return an array with all stakers information
+   * @throws {Error} - An error object if an error occurred, results otherwise
+   */
+  public async getAllStakers(): Promise<IStaker[]> {
+    try {
+      const result = await this.stakingContract.getListOfStakers();
+
+      if (result[1].length === 0) {
+        throw ErrorStakingStakersNotFound;
+      }
+
+      return result[1];
+    } catch (e) {
+      console.error(e);
+      throw ErrorStakingStakersNotFound;
+    }
+  }
+
+  /**
+   * **Returns information about the allocation of the specified escrow.*
+   *
+   * @param {string} escrowAddress - The escrow address for the received allocation data
+   * @returns {Promise<IAllocation>} - Returns allocation info if exists
+   * @throws {Error} - An error object if an error occurred, result otherwise
+   */
+  public async getAllocation(escrowAddress: string): Promise<IAllocation> {
+    if (!ethers.utils.isAddress(escrowAddress)) {
+      throw ErrorInvalidEscrowAddressProvided;
+    }
+
+    try {
+      const result = await this.stakingContract.getAllocation(escrowAddress);
+      return result;
+    } catch (e) {
+      console.error(e);
+      throw ErrorStakingGetAllocation;
+    }
+  }
+
+  /**
+   * **Returns information about the rewards for a given escrow address.*
+   *
+   * @param {string} escrowAddress - Address of the escrow
+   * @returns {Promise<IReward[]>} - Returns rewards info if exists
+   * @throws {Error} - An error object if an error occurred, results otherwise
+   */
+  public async getRewards(escrowAddress: string): Promise<IReward[]> {
```

Comment:
about this method, the idea was to get all rewards for a given slasher, I know the contract itself doesn't allow to do it, but we need to use subgraph for it